modVM/src/org/aion/vm/BulkExecutor.java

```diff
@@ -343,25 +345,25 @@ private void updateRepository(
             RepositoryCache<AccountState, IBlockStoreBase<?, ?>> track =
                     this.repositoryChild.startTracking();
 
+            tx.setNrgConsume(computeEnergyUsed(tx.getEnergyLimit(), result));
+
             // Refund energy if transaction was successfully or reverted.
-            if (result.getResultCode().isSuccess() || result.getResultCode().isRevert()) {
-                if (!transactionIsForAionVirtualMachine(tx)) {
+            if (!transactionIsForAionVirtualMachine(tx)) {
+                if (result.getResultCode().isSuccess() || result.getResultCode().isRevert()) {
                     track.addBalance(tx.getSenderAddress(), summary.getRefund());
                 }
-            }
 
-            tx.setNrgConsume(computeEnergyUsed(tx.getEnergyLimit(), result));
```

Comment:
where is this done now?

Reply:
It just moved up a few lines, before the if statement, since it is not conditional